apps/consumer-client/pages/examples/add-pcd.tsx

```diff
@@ -62,11 +74,201 @@ export default function Page() {
         <button onClick={addWebAuthnPCD}>
           add a new webauthn credential to the passport
         </button>
+        <br />
+        <br />
+        <AddEthAddrPCDButton />
+        <br />
+        <br />
+        <AddEthGroupPCDButton />
       </ExampleContainer>
     </div>
   );
 }
 
+function AddEthAddrPCDButton() {
+  const [pcdStr] = usePassportPopupMessages();
+  const [isActive, setIsActive] = useState(false);
+
+  useEffect(() => {
+    if (!pcdStr) return;
+    if (!isActive) return;
+
+    const parsed = JSON.parse(pcdStr) as SerializedPCD;
+
+    const ethereum = (window as any).ethereum;
+    const provider = new ethers.providers.Web3Provider(ethereum);
+    if (!ethereum) {
+      alert("Please install MetaMask to use this dApp!");
+    }
+
+    (async function () {
+      await ethereum.request({ method: "eth_requestAccounts" });
+      const pcd = await SemaphoreSignaturePCDPackage.deserialize(parsed.pcd);
+      const signature = await provider
+        .getSigner()
+        .signMessage(pcd.claim.identityCommitment);
+
+      const popupUrl = window.location.origin + "/popup";
+
+      const proofUrl = constructPassportPcdProveAndAddRequestUrl<
+        typeof EthereumOwnershipPCDPackage
+      >(PASSPORT_URL, popupUrl, EthereumOwnershipPCDPackage.name, {
+        identity: {
+          argumentType: ArgumentTypeName.PCD,
+          pcdType: SemaphoreIdentityPCDPackage.name,
+          value: undefined,
+          userProvided: true,
+          description:
+            "The Semaphore Identity which you are signing the message.",
+        },
+        ethereumAddress: {
+          argumentType: ArgumentTypeName.String,
+          value: await provider.getSigner().getAddress(),
+        },
+        ethereumSignatureOfCommitment: {
+          argumentType: ArgumentTypeName.String,
+          value: signature,
+        },
+      });
+
+      sendPassportRequest(proofUrl);
+    })();
+
+    setIsActive(false);
+  }, [pcdStr, isActive]);
+
+  return (
+    <button
+      onClick={() => {
+        setIsActive(true);
+        zupassSignIn("eth-pcd");
+      }}
+    >
+      add a new Ethereum address to the passport
+    </button>
+  );
+}
+
+async function zupassSignIn(originalSiteName: string) {
+  openSignedZuzaluSignInPopup(
+    PASSPORT_URL,
+    window.location.origin + "/popup",
+    originalSiteName
+  );
+}
+
+/**
+ * The offset that the hex representation of the public key starts at, without the 0x prefix and without the 04 prefix. That's what the zk-circuit expects.
+ * https://github.com/indutny/elliptic/issues/86
+ * https://dev.to/q9/finally-understanding-ethereum-accounts-1kpe
+ */
+const hexPubkeyOffset = 2 + 2;
+
+function AddEthGroupPCDButton() {
```

Comment:
A good chunk of code here is duplicated in `EthereumGroupPCD.spec.ts`. I tried to refactor it and put this example code in `EthereumGroupPCD.ts` but ran into problems in `EthereumGroupPCD.init()`. I'm leaving it like this for your review.